question.php

```diff
@@ -277,8 +277,10 @@ public function is_complete_response(array $response): bool {
      * @return bool whether the two sets of responses are the same - that is
      *                            whether the new set of responses can safely be discarded.
      */
-    public function is_same_response(array $prevresponse, array $newresponse) {
-        return false;
+    public function is_same_response(array $prevresponse, array $newresponse): bool {
+        /* The response has already been filtered against get_expected_data, we just need to filter out attempt state
+           and scoring state before comparing. */
+        return $newresponse == array_filter($prevresponse, fn($key) => !str_starts_with($key, "_"), ARRAY_FILTER_USE_KEY);
```

Comment:
Im docstring der Funktion steht, dass `$prevresponse` und `$newresponse` im selben Format übergeben werden. Warum müssen dann nur in `$prevresponse` Einträge beginnend mit `'_'` entfernt werden?

Reply:
Da dachte ich naiv bzw. unaufmerksam, dass `$newresponse` aus den `$POST` kommt. War beim testen auch ohne `_`s, aber das war dann Zufall (Bzw. Folge dessen, dass wir den attempt state nur in den ersten Step schreiben).